tests/analyses/test_api.py

```diff
@@ -344,3 +344,42 @@ async def test_blast(error, mocker, spawn_client, resp_is, static_time):
         5,
         "FOOBAR1337"
     )
+
+
+@pytest.mark.parametrize("error", [None, 422, 404, 409])
+async def test_patch_to_set_result(spawn_client, error, resp_is):
+    analysis_document = dict(
+        _id="analysis1",
+        sample=dict(id="sample1"),
+        workflow="test_workflow",
+    )
```

Comment:
Please use literals for creating dictionaries that will be used in Mongo operations. They are easier to read and easily mentally map to JSON.